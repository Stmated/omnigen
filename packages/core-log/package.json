{
  "name": "@omnigen/core-log",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "pino": "8.6.0"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "colors": "^1.4.0",
    "typescript": "^5.5.0-beta",
    "pino-pretty": "^10.3.1"
  }
}
