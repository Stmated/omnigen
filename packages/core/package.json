{
  "name": "@omnigen/core",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@omnigen/core-log": "workspace:*",
    "@etclabscore/string-json-template-language": "^1.0.0",
    "@open-rpc/schema-utils-js": "^1.16.1",
    "change-case": "^4.1.2",
    "deepmerge": "^4.3.0",
    "js-yaml": "^4.1.0",
    "json-pointer": "^0.6.2",
    "node-fetch": "^2.6.9",
    "object-hash": "^3.0.0",
    "string-similarity": "^4.0.4",
    "ts-json-schema-generator": "^1.2.0",
    "zod": "^3.22.4"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "@swc/core": "^1.3.32",
    "@types/json-pointer": "^1.0.31",
    "@types/json-schema": "^7.0.15",
    "@types/node": "^18.13.0",
    "@types/node-fetch": "^2.6.2",
    "@types/object-hash": "^2.2.1",
    "@types/string-similarity": "^4.0.0",
    "colors": "^1.4.0",
    "nock": "^13.3.0",
    "typescript": "^5.4.2",
    "type-fest": "^3.6.1"
  }
}
