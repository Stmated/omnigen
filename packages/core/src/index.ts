import {PluginAutoRegistry} from '@omnigen/core-plugin';
import {CommonTransformPlugin, CommonTransform2Plugin, fileWriter, CorePlugin} from './CoreUtilPluginInit.ts';

export * from './ast';
export * from './equality';
export * from './parse';
export * from './util';
export * from './visit';
export * from './write';
export * from './CoreUtilPluginInit.ts';

export default PluginAutoRegistry.register([CorePlugin, CommonTransformPlugin, CommonTransform2Plugin, fileWriter]);
