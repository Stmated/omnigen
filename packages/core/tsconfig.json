{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@omnigen/config-typescript/build.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "baseUrl": "./src"
  }
}
