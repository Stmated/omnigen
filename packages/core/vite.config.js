import {resolve} from 'path';
import {defineConfig} from 'vite';

export default defineConfig({
  build: {
    outDir: 'dist/lib',
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      // make sure to externalize deps that shouldn't be bundled into your library
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up', 'fs/promises', 'fs'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
      treeshake: 'safest',
    },
  },
  optimizeDeps: {
    exclude: ['sync-fetch'],
  },
});
