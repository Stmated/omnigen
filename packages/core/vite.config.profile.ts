import {resolve} from 'path';
import {type UserConfig} from 'vite';

export default {
  build: {
    outDir: 'dist/profile',
    minify: false,
    terserOptions: {
      compress: false,
      mangle: false,
    },
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.profiling.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
    },
  },
} satisfies UserConfig;
