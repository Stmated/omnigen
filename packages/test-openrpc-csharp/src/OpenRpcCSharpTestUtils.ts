import {BaseContext, FileContext, TargetContext} from '@omnigen/core-plugin';
import {Util, ZodCompilationUnitsContext} from '@omnigen/core-util';
import {z} from 'zod';
import {PluginManager} from '@omnigen/plugin';
import {OpenRpcPlugins} from '@omnigen/parser-openrpc';
import {CSharpOptions, CSharpPlugins} from '@omnigen/target-csharp';
import {LoggerFactory} from '@omnigen/core-log';
import {ModelTransformOptions, PackageOptions, ParserOptions, TargetOptions} from '@omnigen/core';

const logger = LoggerFactory.create(import.meta.url);

export type AllCSharpOptions = ParserOptions & ModelTransformOptions & PackageOptions & TargetOptions & CSharpOptions;

export class OpenRpcCSharpTestUtils {

  static async render(fileName: string, options?: Partial<AllCSharpOptions>) {

    logger.info(`Rendering with ${OpenRpcPlugins.default} and ${CSharpPlugins.default}`);

    const ctx: BaseContext & FileContext & TargetContext = {
      file: fileName.includes('/') ? fileName : Util.getPathFromRoot(`./packages/parser-openrpc/examples/${fileName}`),
      arguments: {
        ...options,
        target: 'csharp',
      },
    };

    const expected = ZodCompilationUnitsContext;
    const stopAt = ZodCompilationUnitsContext;

    const pm = new PluginManager({includeAuto: true});
    try {
      const execResult = await pm.execute({ctx: ctx, debug: true, stopAt: stopAt});
      const last = execResult.results[execResult.results.length - 1];

      const result = expected.parse(last.ctx) as z.output<typeof expected>;

      return result.compilationUnits;
    } catch (ex) {
      throw LoggerFactory.formatError(ex);
    }
  }
}
