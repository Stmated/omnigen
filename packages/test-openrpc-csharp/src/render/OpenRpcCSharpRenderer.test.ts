import {beforeEach, describe, expect, test, Test, vi} from 'vitest';
import {LoggerFactory} from '@omnigen/core-log';
import {OpenRpcCSharpTestUtils} from '../OpenRpcCSharpTestUtils.ts';
import {RenderedCompilationUnit} from '@omnigen/core';

const logger = LoggerFactory.create(import.meta.url);

describe('OpenRpc+CSharp Rendering', () => {

  beforeEach(() => {
    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});
  });

  function verify(task: Test<{}>, units: RenderedCompilationUnit[]) {
    const fileContents = new Map<string, string>();
    for (const cu of units) {
      fileContents.set(cu.fileName, cu.content);
      expect(cu.content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${cu.fileName}`);
    }
    expect([...fileContents.keys()].sort()).toMatchSnapshot();
  }

  function getFileName(task: Test<{}>): string {

    const taskName = task.name;
    const underscoreIdx = taskName.indexOf('_');
    if (underscoreIdx === -1) {
      return `${taskName}.json`;
    } else {
      return `${taskName.substring(0, underscoreIdx)}.json`;
    }
  }

  test('multiple-inheritance', async ({task}) => verify(task, await OpenRpcCSharpTestUtils.render(getFileName(task), {
    singleFile: true,
  })));

  test('compressable-types', async ({task}) => verify(task, await OpenRpcCSharpTestUtils.render(getFileName(task), {
    generifyTypes: false,
    singleFile: true,
  })));

  test('enum', async ({task}) => verify(task, await OpenRpcCSharpTestUtils.render(getFileName(task), {
    singleFile: true,
    includeGenerated: false,
  })));

  test('additional-properties', async ({task}) => verify(task, await OpenRpcCSharpTestUtils.render(getFileName(task), {
    additionalPropertiesInterfaceAfterDuplicateCount: 1,
    singleFile: true,
  })));

  test('method-in-response', async ({task}) => verify(task, await OpenRpcCSharpTestUtils.render(getFileName(task), {
    // preferInterfaces: true,
    singleFile: true,
  })));
});
