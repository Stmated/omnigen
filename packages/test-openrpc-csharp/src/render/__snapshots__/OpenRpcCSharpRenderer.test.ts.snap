// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OpenRpc+CSharp Rendering > additional-properties 1`] = `
[
  "JsonRpcRequest.cs",
]
`;

exports[`OpenRpc+CSharp Rendering > compressable-types 1`] = `
[
  "JsonRpcRequest.cs",
]
`;

exports[`OpenRpc+CSharp Rendering > enum 1`] = `
[
  "JsonRpcRequest.cs",
]
`;

exports[`OpenRpc+CSharp Rendering > method-in-response 1`] = `
[
  "DepositRequest.cs",
]
`;

exports[`OpenRpc+CSharp Rendering > multiple-inheritance 1`] = `
[
  "GiveInGetOutRequest.cs",
]
`;
